pgconn/config.go

```diff
@@ -467,22 +467,25 @@ func parseEnvSettings() map[string]string {
 func parseURLSettings(connString string) (map[string]string, error) {
 	settings := make(map[string]string)
 
-	url, err := url.Parse(connString)
+	parsedURL, err := url.Parse(connString)
 	if err != nil {
-		return nil, err
+		if urlErr := new(url.Error); errors.As(err, &urlErr) {
+			return nil, urlErr.Err
+		}
+		return nil, errors.New("failed to parse URL")
```

Comment:
```suggestion
		return nil, err
```
for non url.Error should be able to return underlying error

Reply:
I agree with change out of general principal. Though this path will never be reached, url.Parse only returns url.Error.

Reply:
changed